### ISSUE TITLE
Not working through proxy anymore

### ISSUE THREAD

**Issue by chipmunk2000** [GoogleCloudPlatform/iap-desktop#851]
working in version [2.27.810](https://github.com/GoogleCloudPlatform/iap-desktop/releases/tag/2.27.810)

not working anymore in 2_33_964 i didnt try the versions inbetween 

I can still bowse the projects, but when trying to add I get a 407  
Proxy is set up correctly, otherwise I couldnt browse the projects

Upon deinstalling and reinstalling the old version it works again.
Deinstalling  the old version and installing the new version once again gets the system to work for a short period of time

Existing projects, that were accessibale before are being shown as unreachable with the newest version

 
Google Verbose: 0 : Request[00000005] (triesRemaining=3) URI: '[https://cloudresourcemanager.googleapis.com/v1/projects/redacted'](https://cloudresourcemanager.googleapis.com/v1/projects/redacted)
Google Error: 0 : Response[00000004] Exception was thrown while executing a HTTP request and it wasn't handled
Google Error: 0 : System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Net.WebException: The remote server returned an error: (407) Proxy Authentication Required. ---> System.ComponentModel.Win32Exception: The token supplied to the function is invalid
   at System.Net.NTAuthentication.GetOutgoingBlob(Byte[] incomingBlob, Boolean throwOnError, SecurityStatus& statusCode)
   at System.Net.NTAuthentication.GetOutgoingBlob(String incomingBlob)
   at System.Net.NegotiateClient.DoAuthenticate(String challenge, WebRequest webRequest, ICredentials credentials, Boolean preAuthenticate)
   at System.Net.NegotiateClient.Authenticate(String challenge, WebRequest webRequest, ICredentials credentials)
   at System.Net.AuthenticationManagerDefault.Authenticate(String challenge, WebRequest request, ICredentials credentials)
   at System.Net.AuthenticationState.AttemptAuthenticate(HttpWebRequest httpWebRequest, ICredentials authInfo)
   at System.Net.HttpWebRequest.CheckResubmitForAuth()
   at System.Net.HttpWebRequest.CheckResubmit(Exception& e, Boolean& disableUpload)
   --- End of inner exception stack trace ---
   at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
   at System.Net.Http.HttpClientHandler.GetResponseCallback(IAsyncResult ar)
   --- End of inner exception stack trace ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Google.Apis.Http.ConfigurableMessageHandler.<SendAsync>d__69.MoveNext()
Google Error: 0 : Response[00000001] Exception was thrown while executing a HTTP request and it wasn't handled
Google Error: 0 : System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Net.WebException: The remote server returned an error: (407) Proxy Authentication Required. ---> System.ComponentModel.Win32Exception: The token supplied to the function is invalid
   at System.Net.NTAuthentication.GetOutgoingBlob(Byte[] incomingBlob, Boolean throwOnError, SecurityStatus& statusCode)
   at System.Net.NTAuthentication.GetOutgoingBlob(String incomingBlob)
   at System.Net.NegotiateClient.DoAuthenticate(String challenge, WebRequest webRequest, ICredentials credentials, Boolean preAuthenticate)
   at System.Net.NegotiateClient.Authenticate(String challenge, WebRequest webRequest, ICredentials credentials)
   at System.Net.AuthenticationManagerDefault.Authenticate(String challenge, WebRequest request, ICredentials credentials)
   at System.Net.AuthenticationState.AttemptAuthenticate(HttpWebRequest httpWebRequest, ICredentials authInfo)
   at System.Net.HttpWebRequest.CheckResubmitForAuth()
   at System.Net.HttpWebRequest.CheckResubmit(Exception& e, Boolean& disableUpload)
   --- End of inner exception stack trace ---
   at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
   at System.Net.Http.HttpClientHandler.GetResponseCallback(IAsyncResult ar)
   --- End of inner exception stack trace ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Google.Apis.Http.ConfigurableMessageHandler.<SendAsync>d__69.MoveNext()
Google.Solutions.IapDesktop.Application Verbose: 0 : Exit GetProjectAsync()
Google.Solutions.IapDesktop.Application Verbose: 0 : Exit GetProjectAsync()
Google.Solutions.IapDesktop.Application Error: 0 : Failed to load project projects/redacted: System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Net.WebException: The remote server returned an error: (407) Proxy Authentication Required. ---> System.ComponentModel.Win32Exception: The token supplied to the function is invalid
   at System.Net.NTAuthentication.GetOutgoingBlob(Byte[] incomingBlob, Boolean throwOnError, SecurityStatus& statusCode)
   at System.Net.NTAuthentication.GetOutgoingBlob(String incomingBlob)
   at System.Net.NegotiateClient.DoAuthenticate(String challenge, WebRequest webRequest, ICredentials credentials, Boolean preAuthenticate)
   at System.Net.NegotiateClient.Authenticate(String challenge, WebRequest webRequest, ICredentials credentials)
   at System.Net.AuthenticationManagerDefault.Authenticate(String challenge, WebRequest request, ICredentials credentials)
   at System.Net.AuthenticationState.AttemptAuthenticate(HttpWebRequest httpWebRequest, ICredentials authInfo)
   at System.Net.HttpWebRequest.CheckResubmitForAuth()
   at System.Net.HttpWebRequest.CheckResubmit(Exception& e, Boolean& disableUpload)
   --- End of inner exception stack trace ---
   at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
   at System.Net.Http.HttpClientHandler.GetResponseCallback(IAsyncResult ar)
   --- End of inner exception stack trace ---
   at Google.Apis.Http.ConfigurableMessageHandler.<SendAsync>d__69.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Google.Apis.Requests.ClientServiceRequest`1.<ExecuteUnparsedAsync>d__34.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Google.Apis.Requests.ClientServiceRequest`1.<ExecuteAsync>d__31.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Google.Solutions.IapDesktop.Application.Services.Adapters.ResourceManagerAdapter.<GetProjectAsync>d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Google.Solutions.IapDesktop.Application.Services.ProjectModel.ProjectModelService.<LoadProjectsAsync>d__10.MoveNext()
Google Verbose: 0 : Response[00000005] Response status: OK 'OK'

**Comment by jpassing** [comment]
Thanks for reporting this issue.

Version [2.31](https://github.com/GoogleCloudPlatform/iap-desktop/releases/tag/2.31.901) introduced a new _Add/Remove projects_ dialog, maybe the issue is related to that change.

To clarify: 
1. Which of the 3 proxy options under **Tools > Options > Network** are you currently using -- _Use system settings_?
2. Are you using Endpoint Verification (**Tools > Options > General**)? (It's off by default)

**Comment by chipmunk2000** [comment]
Use automatic configuartion script
and proxy server requires authentication ticked  and obiously my credentials :-)

**Comment by jpassing** [comment]
I spent some time trying to reproduce the issue with 3 different proxy configurations:

1. No authentication
2. NTLM authentication
3. Kerberos authentication

But everything worked fine. 

To diagnose the issue further, it would be great if you could enable diagnostic logs for the .NET runtime:

1. Open the folder `%APPDATA%\Google\IAP Desktop\`
2. Create a sub-folder `Logs` (if it doesn't exist yet)
3. Create a copy of the file `iapdesktop.exe.config` (as a backup)
4. Replace `iapdesktop.exe.config` with [this file](https://gist.githubusercontent.com/jpassing/28e0bab41908f4101e93465263a385bc/raw/479414df10ddca2c15f4c5c2483ae4c8977f4932/IapDesktop.exe.config)
6. Restart IAP Desktop and reproduce the error
7. Restore your original `iapdesktop.exe.config` 

The resulting log file `Logs\system.net.log` should hopefully contain more error details. Instead of posting it here, you can also send it to iap-desktop@google.com.

Thanks for your help!

**Comment by chipmunk2000** [comment]
Now it doesnt work at all anymore
When starting it prompts me to sign into foole and tells me after depressing the button IAP Destop failed to complete the OAUTH authorization ... again a 407 error


contents of systen.net.log ==> I had to change the hostnames and IP'S and realm for obvious reasons :-)

System.Net Information: 0 : [16804] Der aktuelle Betriebssysteminstallationstyp ist "Client".
System.Net Information: 0 : [16804] RAS unterstützt: True
System.Net.Http Verbose: 0 : [66412] Entering HttpClientHandler#55416443::.ctor()
System.Net.Http Verbose: 0 : [66412] Exiting HttpClientHandler#55416443::.ctor() 
System.Net.Http Verbose: 0 : [66412] Entering StreamInterceptionHandler#2555439::.ctor()
System.Net.Http Verbose: 0 : [66412] Exiting StreamInterceptionHandler#2555439::.ctor() 
System.Net.Http Information: 0 : [66412] Associating StreamInterceptionHandler#2555439 with HttpClientHandler#55416443
System.Net.Http Verbose: 0 : [66412] Entering ConfigurableMessageHandler#43442467::.ctor()
System.Net.Http Verbose: 0 : [66412] Exiting ConfigurableMessageHandler#43442467::.ctor() 
System.Net.Http Information: 0 : [66412] Associating ConfigurableMessageHandler#43442467 with StreamInterceptionHandler#2555439
System.Net.Http Verbose: 0 : [66412] Entering ConfigurableHttpClient#324444::.ctor(ConfigurableMessageHandler#43442467)
System.Net.Http Information: 0 : [66412] Associating ConfigurableHttpClient#324444 with ConfigurableMessageHandler#43442467
System.Net.Http Verbose: 0 : [66412] Exiting ConfigurableHttpClient#324444::.ctor() 
System.Net.Http Verbose: 0 : [66412] Entering ConfigurableHttpClient#324444::.ctor(ConfigurableMessageHandler#43442467)
System.Net.Http Verbose: 0 : [66412] Exiting ConfigurableHttpClient#324444::.ctor() 
System.Net.Http Verbose: 0 : [66412] Entering HttpClientHandler#8198134::.ctor()
System.Net.Http Verbose: 0 : [66412] Exiting HttpClientHandler#8198134::.ctor() 
System.Net.Http Verbose: 0 : [66412] Entering HttpClient#5150550::.ctor(HttpClientHandler#8198134)
System.Net.Http Information: 0 : [66412] Associating HttpClient#5150550 with HttpClientHandler#8198134
System.Net.Http Verbose: 0 : [66412] Exiting HttpClient#5150550::.ctor() 
System.Net.Http Verbose: 0 : [66412] Entering HttpClient#5150550::.ctor(HttpClientHandler#8198134)
System.Net.Http Verbose: 0 : [66412] Exiting HttpClient#5150550::.ctor() 
System.Net.Http Verbose: 0 : [66412] Entering HttpRequestMessage#20450490::.ctor(Method: GET, Uri: 'https://openidconnect.googleapis.com/v1/userinfo')
System.Net.Http Verbose: 0 : [66412] Exiting HttpRequestMessage#20450490::.ctor() 
System.Net.Http Verbose: 0 : [45600] Entering HttpRequestMessage#4032828::.ctor(Method: POST, Uri: 'https://oauth2.googleapis.com/token')
System.Net.Http Verbose: 0 : [45600] Exiting HttpRequestMessage#4032828::.ctor() 
System.Net.Http Verbose: 0 : [45600] Entering FormUrlEncodedContent#33711845::.ctor()
System.Net.Http Verbose: 0 : [45600] Exiting FormUrlEncodedContent#33711845::.ctor() 
System.Net.Http Information: 0 : [45600] Associating HttpRequestMessage#4032828 with FormUrlEncodedContent#33711845
System.Net.Http Verbose: 0 : [45600] Entering ConfigurableHttpClient#324444::SendAsync(HttpRequestMessage#4032828: Method: POST, RequestUri: 'https://oauth2.googleapis.com/token', Version: 1.1, Content: System.Net.Http.FormUrlEncodedContent, Headers:
{
  Content-Type: application/x-www-form-urlencoded
})
System.Net.Http Verbose: 0 : [45600] Entering HttpClientHandler#55416443::SendAsync(HttpRequestMessage#4032828)
System.Net Verbose: 0 : [45600] Entering HttpWebRequest#37489757::HttpWebRequest(https://oauth2.googleapis.com/token#-1749135709)
System.Net Verbose: 0 : [45600] Exiting HttpWebRequest#37489757::HttpWebRequest() 
System.Net Verbose: 0 : [45600] Entering HttpWebRequest#37489757::HttpWebRequest(uri: 'https://oauth2.googleapis.com/token', connectionGroupName: '55416443')
System.Net Verbose: 0 : [45600] Exiting HttpWebRequest#37489757::HttpWebRequest() 
System.Net.Http Information: 0 : [45600] Associating HttpRequestMessage#4032828 with HttpWebRequest#37489757
System.Net Verbose: 0 : [45600] Entering ServicePoint#64828693::ServicePoint(pxvip02.a.domain:8080)
System.Net Information: 0 : [45600] Associating HttpWebRequest#37489757 with ServicePoint#64828693
System.Net.Http Verbose: 0 : [45600] Exiting HttpClientHandler#55416443::SendAsync() 	-> Task`1#10104599
System.Net Verbose: 0 : [48416] Entering HttpWebRequest#37489757::BeginGetRequestStream()
System.Net Information: 0 : [48416] Associating Connection#12549444 with HttpWebRequest#37489757
System.Net Verbose: 0 : [48416] Entering HttpWebRequest#11478475::HttpWebRequest(http://pxvip02.a.domain:8080/#1479616988)
System.Net Verbose: 0 : [48416] Exiting HttpWebRequest#11478475::HttpWebRequest() 
System.Net.Http Verbose: 0 : [45600] Exiting ConfigurableHttpClient#324444::SendAsync() 	-> Task`1#51288387
System.Net Verbose: 0 : [48416] Entering ServicePoint#43245362::ServicePoint(pxvip02.a.domain:8080)
System.Net Information: 0 : [48416] Associating HttpWebRequest#11478475 with ServicePoint#43245362
System.Net Verbose: 0 : [48416] Entering HttpWebRequest#11478475::BeginGetResponse()
System.Net Information: 0 : [48416] Associating Connection#17810826 with HttpWebRequest#11478475
System.Net Verbose: 0 : [48416] Exiting HttpWebRequest#11478475::BeginGetResponse() 	-> ContextAwareResult#50886371
System.Net Verbose: 0 : [48416] Exiting HttpWebRequest#37489757::BeginGetRequestStream() 	-> ContextAwareResult#1528652
System.Net Information: 0 : [63404] Connection#17810826 - Die Verbindung von 10.80.6.44:59091 mit 10.10.20.9:8080 wurde hergestellt.
System.Net Information: 0 : [63404] Associating HttpWebRequest#11478475 with ConnectStream#33156464
System.Net Information: 0 : [63404] HttpWebRequest#11478475 - Request: CONNECT oauth2.googleapis.com:443 HTTP/1.1

System.Net Information: 0 : [63404] ConnectStream#33156464 - Header
{
User-Agent: google-api-dotnet-client/1.57.0.0 (gzip)
Host: oauth2.googleapis.com
Proxy-Connection: Keep-Alive
} werden gesendet.
System.Net Information: 0 : [18252] Connection#17810826 - Empfangene Statusleiste: Version = 1.1, StatusCode = 407, StatusDescription = Proxy Authentication Required.
System.Net Information: 0 : [18252] Connection#17810826 - Header
{
Proxy-Authenticate: NEGOTIATE,NTLM,BASIC realm="myrealm"
Cache-Control: no-cache
Pragma: no-cache
Content-Type: text/html; charset=utf-8
Proxy-Connection: close
Connection: close
Content-Length: 797
} wurden empfangen.
System.Net Information: 0 : [18252] ConnectStream#7995840::ConnectStream(Es wurden 797 Bytes gepuffert.)
System.Net Information: 0 : [18252] Associating HttpWebRequest#11478475 with ConnectStream#7995840
System.Net Information: 0 : [18252] Associating HttpWebRequest#11478475 with HttpWebResponse#56251872
System.Net Information: 0 : [18252] Sicherheitspakete werden enumeriert:
System.Net Information: 0 : [18252]     Negotiate
System.Net Information: 0 : [18252]     NegoExtender
System.Net Information: 0 : [18252]     Kerberos
System.Net Information: 0 : [18252]     NTLM
System.Net Information: 0 : [18252]     TSSSP
System.Net Information: 0 : [18252]     pku2u
System.Net Information: 0 : [18252]     WDigest
System.Net Information: 0 : [18252]     Schannel
System.Net Information: 0 : [18252]     Microsoft Unified Security Protocol Provider
System.Net Information: 0 : [18252]     Default TLS SSP
System.Net Information: 0 : [18252]     CREDSSP
System.Net Information: 0 : [18252] AcquireCredentialsHandle(package  = Negotiate, intent   = Outbound, authdata = System.Net.SafeSspiAuthDataHandle)
System.Net Information: 0 : [18252] InitializeSecurityContext(credential = System.Net.SafeFreeCredential_SECURITY, context = (null), targetName = HTTP/pxvip02.a.domain, inFlags = Delegate, MutualAuth, Connection)
System.Net Information: 0 : [18252] InitializeSecurityContext(Anzahl von In-Buffers = 0, Länge von Out-Buffer = 40, zurückgegebener Code = ContinueNeeded).
System.Net Warning: 0 : [18252] HttpWebRequest#11478475::() - Anforderung wird erneut übertragen.
System.Net Information: 0 : [18252] Associating HttpWebRequest#11478475 with ServicePoint#43245362
System.Net Information: 0 : [18252] Associating Connection#4878312 with HttpWebRequest#11478475
System.Net Information: 0 : [18252] Connection#4878312 - Die Verbindung von 10.80.6.44:59092 mit 10.10.20.9:8080 wurde hergestellt.
System.Net Information: 0 : [18252] Associating HttpWebRequest#11478475 with ConnectStream#37710717
System.Net Information: 0 : [18252] HttpWebRequest#11478475 - Request: CONNECT oauth2.googleapis.com:443 HTTP/1.1

System.Net Information: 0 : [18252] ConnectStream#37710717 - Header
{
User-Agent: google-api-dotnet-client/1.57.0.0 (gzip)
Proxy-Authorization: Negotiate TlRMTVNTUAABAAAAl4II4gAAAAAAAAAAAAAAAAAAAAAKAGFKAAAADw==
Host: oauth2.googleapis.com
} werden gesendet.
System.Net Information: 0 : [66180] Connection#4878312 - Empfangene Statusleiste: Version = 1.1, StatusCode = 407, StatusDescription = Proxy Authentication Required.
System.Net Information: 0 : [66180] Connection#4878312 - Header
{
Pragma: no-cache
Proxy-Connection: Keep-Alive
Connection: Keep-Alive
Content-Length: 814
Cache-Control: no-cache
Content-Type: text/html; charset=utf-8
Proxy-Authenticate: NEGOTIATE TlRMTVNTUAACAAAABgAGADgAAAAVAongLZiUbVpGNSMAAAAAAAAAAJQAlAA+AAAABQCTCAAAAA9aAFQAQgACAAYAWgBUAEIAAQAOAFAAWABGAEYATQAyADQABAAuAFoAVABCAC4ASQBDAEIALgBDAE8ATQBNAEUAUgBaAEIAQQBOAEsALgBDAE8ATQADAD4AcAB4AGYAZgBtADIANAAuAHoAdABiAC4AaQBjAGIALgBjAG8AbQBtAGUAcgB6AGIAYQBuAGsALgBjAG8AbQAAAAAA
} wurden empfangen.
System.Net Information: 0 : [66180] ConnectStream#36610825::ConnectStream(Es wurden 814 Bytes gepuffert.)
System.Net Information: 0 : [66180] Associating HttpWebRequest#11478475 with ConnectStream#36610825
System.Net Information: 0 : [66180] Associating HttpWebRequest#11478475 with HttpWebResponse#61325552
System.Net Information: 0 : [66180] InitializeSecurityContext(credential = System.Net.SafeFreeCredential_SECURITY, context = a995f48:5d13058, targetName = HTTP/pxvip02.a.domain, inFlags = Delegate, MutualAuth, Connection)
System.Net Information: 0 : [66180] InitializeSecurityContext(Anzahl von In-Buffers = 1, Länge von Out-Buffer = 504, zurückgegebener Code = OK).
System.Net Warning: 0 : [66180] HttpWebRequest#11478475::() - Anforderung wird erneut übertragen.
System.Net Information: 0 : [66180] Associating HttpWebRequest#11478475 with ServicePoint#43245362
System.Net Information: 0 : [66180] Associating Connection#4878312 with HttpWebRequest#11478475
System.Net Information: 0 : [66180] Associating HttpWebRequest#11478475 with ConnectStream#29028787
System.Net Information: 0 : [66180] HttpWebRequest#11478475 - Request: CONNECT oauth2.googleapis.com:443 HTTP/1.1

System.Net Information: 0 : [66180] ConnectStream#29028787 - Header
{
User-Agent: google-api-dotnet-client/1.57.0.0 (gzip)
Proxy-Authorization: Negotiate TlRMTVNTUAADAAAAGAAYAHYAAABaAVoBjgAAAAAAAABYAAAADgAOAFgAAAAQABAAZgAAABAAEADoAQAAFQKI4goAYUoAAAAPreiO/nauYe2W1tuCSb2WzGMAYgAyAGwAbwBiAHIAQQBQADkAMgA3ADkANQAyAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEltDo7SXbsJs+jvGCapGGwBAQAAAAAAAFK4wV2bE9kBJGiXZ+vkE5zLb/98AgAGAFoAVABCAAEADgBQAFgARgBGAE0AMgA0AAQALgBaAFQAQgAuAEkAQwBCAC4AQwBPAE0ATQBFAFIAWgBCAEEATgBLAC4AQwBPAE0AAwA+AHAAeABmAGYAbQAyADQALgB6AHQAYgAuAGkAYwBiAC4AYwBvAG0AbQBlAHIAegBiAGEAbgBrAC4AYwBvAG0ACAAwADAAAAAAAAAAAAAAAAAgAABADAG42Xfm+XxzC2p2aMxtS+0HUfhw2iIHgfXTWXPnqwoAEAAAAAAAAAAAAAAAAAAAAAAACQBKAEgAVABUAFAALwBwAHgAdgBpAHAAMAAyAC4AaQBuAHQAcgBhAG4AZQB0AC4AYwBvAG0AbQBlAHIAegBiAGEAbgBrAC4AYwBvAG0AAAAAAFNw5qRSVPsOjALzdi4OmFnOy7l8
Host: oauth2.googleapis.com
} werden gesendet.
System.Net Information: 0 : [18252] Connection#4878312 - Empfangene Statusleiste: Version = 1.1, StatusCode = 407, StatusDescription = Proxy Authentication Required.
System.Net Information: 0 : [18252] Connection#4878312 - Header
{
Proxy-Authenticate: NEGOTIATE,NTLM
Cache-Control: no-cache
Pragma: no-cache
Content-Type: text/html; charset=utf-8
Proxy-Connection: close
Connection: close
Content-Length: 847
} wurden empfangen.
System.Net Information: 0 : [18252] ConnectStream#35885827::ConnectStream(Es wurden 847 Bytes gepuffert.)
System.Net Information: 0 : [18252] Associating HttpWebRequest#11478475 with ConnectStream#35885827
System.Net Information: 0 : [18252] Associating HttpWebRequest#11478475 with HttpWebResponse#13575069
System.Net Information: 0 : [18252] ServicePoint#43245362::ReleaseConnectionGroup(/.NET/NetClasses/HttpWebRequest/CONNECT__Group$$/(1)S>I>System.Net.NegotiateClient/2)
System.Net Information: 0 : [18252] ServicePoint#43245362::ReleaseConnectionGroup, returning(true)
System.Net Verbose: 0 : [18252] Entering ConnectStream#35885827::Read()
System.Net Verbose: 0 : [18252] Data from ConnectStream#35885827::Read
System.Net Verbose: 0 : [18252] 00000000 : 3C 48 54 4D 4C 3E 3C 48-45 41 44 3E 0D 0A 3C 54 : <HTML><HEAD>..<T
System.Net Verbose: 0 : [18252] 00000010 : 49 54 4C 45 3E 41 63 63-65 73 73 20 44 65 6E 69 : ITLE>Access Deni
System.Net Verbose: 0 : [18252] 00000020 : 65 64 3C 2F 54 49 54 4C-45 3E 0D 0A 3C 2F 48 45 : ed</TITLE>..</HE
System.Net Verbose: 0 : [18252] 00000030 : 41 44 3E 0D 0A 3C 42 4F-44 59 3E 0D 0A 3C 62 6F : AD>..<BODY>..<bo
System.Net Verbose: 0 : [18252] 00000040 : 64 79 20 62 67 63 6F 6C-6F 72 3D 22 23 46 46 46 : dy bgcolor="#FFF
System.Net Verbose: 0 : [18252] 00000050 : 46 45 41 22 3E 0D 0A 3C-62 6C 6F 63 6B 71 75 6F : FEA">..<blockquo
System.Net Verbose: 0 : [18252] 00000060 : 74 65 3E 0D 0A 3C 54 41-42 4C 45 20 62 6F 72 64 : te>..<TABLE bord
System.Net Verbose: 0 : [18252] 00000070 : 65 72 3D 30 20 63 65 6C-6C 50 61 64 64 69 6E 67 : er=0 cellPadding
System.Net Verbose: 0 : [18252] 00000080 : 3D 31 20 77 69 64 74 68-3D 22 38 30 25 22 3E 0D : =1 width="80%">.
System.Net Verbose: 0 : [18252] 00000090 : 0A 3C 54 52 3E 3C 54 44-3E 0D 0A 3C 46 4F 4E 54 : .<TR><TD>..<FONT
System.Net Verbose: 0 : [18252] 000000A0 : 20 66 61 63 65 3D 22 56-65 72 64 61 6E 61 22 3E :  face="Verdana">
System.Net Verbose: 0 : [18252] 000000B0 : 0D 0A 3C 62 69 67 3E 41-63 63 65 73 73 20 44 65 : ..<big>Access De
System.Net Verbose: 0 : [18252] 000000C0 : 6E 69 65 64 20 28 61 75-74 68 65 6E 74 69 63 61 : nied (authentica
System.Net Verbose: 0 : [18252] 000000D0 : 74 69 6F 6E 5F 66 61 69-6C 65 64 29 3C 2F 62 69 : tion_failed)</bi
System.Net Verbose: 0 : [18252] 000000E0 : 67 3E 0D 0A 3C 42 52 3E-0D 0A 3C 42 52 3E 0D 0A : g>..<BR>..<BR>..
System.Net Verbose: 0 : [18252] 000000F0 : 3C 2F 46 4F 4E 54 3E 0D-0A 3C 2F 54 44 3E 3C 2F : </FONT>..</TD></
System.Net Verbose: 0 : [18252] 00000100 : 54 52 3E 0D 0A 3C 54 52-3E 3C 54 44 3E 0D 0A 3C : TR>..<TR><TD>..<
System.Net Verbose: 0 : [18252] 00000110 : 46 4F 4E 54 20 66 61 63-65 3D 22 56 65 72 64 61 : FONT face="Verda
System.Net Verbose: 0 : [18252] 00000120 : 6E 61 22 3E 0D 0A 59 6F-75 72 20 63 72 65 64 65 : na">..Your crede
System.Net Verbose: 0 : [18252] 00000130 : 6E 74 69 61 6C 73 20 63-6F 75 6C 64 20 6E 6F 74 : ntials could not
System.Net Verbose: 0 : [18252] 00000140 : 20 62 65 20 61 75 74 68-65 6E 74 69 63 61 74 65 :  be authenticate
System.Net Verbose: 0 : [18252] 00000150 : 64 3A 20 22 47 65 6E 65-72 61 6C 20 61 75 74 68 : d: "General auth
System.Net Verbose: 0 : [18252] 00000160 : 65 6E 74 69 63 61 74 69-6F 6E 20 66 61 69 6C 75 : entication failu
System.Net Verbose: 0 : [18252] 00000170 : 72 65 20 64 75 65 20 74-6F 20 62 61 64 20 75 73 : re due to bad us
System.Net Verbose: 0 : [18252] 00000180 : 65 72 20 49 44 20 6F 72-20 61 75 74 68 65 6E 74 : er ID or authent
System.Net Verbose: 0 : [18252] 00000190 : 69 63 61 74 69 6F 6E 20-74 6F 6B 65 6E 2E 22 2E : ication token.".
System.Net Verbose: 0 : [18252] 000001A0 : 20 59 6F 75 20 77 69 6C-6C 20 6E 6F 74 20 62 65 :  You will not be
System.Net Verbose: 0 : [18252] 000001B0 : 20 70 65 72 6D 69 74 74-65 64 20 61 63 63 65 73 :  permitted acces
System.Net Verbose: 0 : [18252] 000001C0 : 73 20 75 6E 74 69 6C 20-79 6F 75 72 20 63 72 65 : s until your cre
System.Net Verbose: 0 : [18252] 000001D0 : 64 65 6E 74 69 61 6C 73-20 63 61 6E 20 62 65 20 : dentials can be 
System.Net Verbose: 0 : [18252] 000001E0 : 76 65 72 69 66 69 65 64-2E 0D 0A 3C 2F 46 4F 4E : verified...</FON
System.Net Verbose: 0 : [18252] 000001F0 : 54 3E 0D 0A 3C 2F 54 44-3E 3C 2F 54 52 3E 0D 0A : T>..</TD></TR>..
System.Net Verbose: 0 : [18252] 00000200 : 3C 54 52 3E 3C 54 44 3E-0D 0A 3C 46 4F 4E 54 20 : <TR><TD>..<FONT 
System.Net Verbose: 0 : [18252] 00000210 : 66 61 63 65 3D 22 56 65-72 64 61 6E 61 22 3E 0D : face="Verdana">.
System.Net Verbose: 0 : [18252] 00000220 : 0A 54 68 69 73 20 69 73-20 74 79 70 69 63 61 6C : .This is typical
System.Net Verbose: 0 : [18252] 00000230 : 6C 79 20 63 61 75 73 65-64 20 62 79 20 61 6E 20 : ly caused by an 
System.Net Verbose: 0 : [18252] 00000240 : 69 6E 63 6F 72 72 65 63-74 20 75 73 65 72 6E 61 : incorrect userna
System.Net Verbose: 0 : [18252] 00000250 : 6D 65 20 61 6E 64 2F 6F-72 20 70 61 73 73 77 6F : me and/or passwo
System.Net Verbose: 0 : [18252] 00000260 : 72 64 2C 20 62 75 74 20-63 6F 75 6C 64 20 61 6C : rd, but could al
System.Net Verbose: 0 : [18252] 00000270 : 73 6F 20 62 65 20 63 61-75 73 65 64 20 62 79 20 : so be caused by 
System.Net Verbose: 0 : [18252] 00000280 : 6E 65 74 77 6F 72 6B 20-70 72 6F 62 6C 65 6D 73 : network problems
System.Net Verbose: 0 : [18252] 00000290 : 2E 0D 0A 3C 2F 46 4F 4E-54 3E 0D 0A 3C 2F 54 44 : ...</FONT>..</TD
System.Net Verbose: 0 : [18252] 000002A0 : 3E 3C 2F 54 52 3E 0D 0A-3C 54 52 3E 3C 54 44 3E : ></TR>..<TR><TD>
System.Net Verbose: 0 : [18252] 000002B0 : 0D 0A 3C 46 4F 4E 54 20-66 61 63 65 3D 22 56 65 : ..<FONT face="Ve
System.Net Verbose: 0 : [18252] 000002C0 : 72 64 61 6E 61 22 20 53-49 5A 45 3D 32 3E 0D 0A : rdana" SIZE=2>..
System.Net Verbose: 0 : [18252] 000002D0 : 3C 42 52 3E 0D 0A 46 6F-72 20 61 73 73 69 73 74 : <BR>..For assist
System.Net Verbose: 0 : [18252] 000002E0 : 61 6E 63 65 2C 20 63 6F-6E 74 61 63 74 20 79 6F : ance, contact yo
System.Net Verbose: 0 : [18252] 000002F0 : 75 72 20 54 65 63 68 20-44 65 73 6B 20 2E 20 50 : ur Tech Desk . P
System.Net Verbose: 0 : [18252] 00000300 : 58 46 46 4D 32 34 0D 0A-3C 2F 46 4F 4E 54 3E 0D : XFFM24..</FONT>.
System.Net Verbose: 0 : [18252] 00000310 : 0A 3C 2F 54 44 3E 3C 2F-54 52 3E 0D 0A 3C 2F 54 : .</TD></TR>..</T
System.Net Verbose: 0 : [18252] 00000320 : 41 42 4C 45 3E 0D 0A 3C-2F 62 6C 6F 63 6B 71 75 : ABLE>..</blockqu
System.Net Verbose: 0 : [18252] 00000330 : 6F 74 65 3E 0D 0A 3C 2F-46 4F 4E 54 3E 0D 0A 3C : ote>..</FONT>..<
System.Net Verbose: 0 : [18252] 00000340 : 2F 42 4F 44 59 3E 3C 2F-48 54 4D 4C 3E 0D 0A    : /BODY></HTML>..
System.Net Verbose: 0 : [18252] Exiting ConnectStream#35885827::Read() 	-> Int32#847
System.Net Verbose: 0 : [18252] Entering ConnectStream#35885827::Read()
System.Net Verbose: 0 : [18252] Exiting ConnectStream#35885827::Read() 	-> Int32#0
System.Net Verbose: 0 : [18252] Entering HttpWebRequest#11478475::EndGetResponse()
System.Net Error: 0 : [18252] Ausnahme in HttpWebRequest#11478475::EndGetResponse - Der Remoteserver hat einen Fehler zurückgegeben: (407) Proxyauthentifizierung erforderlich..
System.Net Error: 0 : [18252] Ausnahme in HttpWebRequest#37489757:: - Der Remoteserver hat einen Fehler zurückgegeben: (407) Proxyauthentifizierung erforderlich..
   bei System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
   bei System.Net.Connection.TunnelThroughProxyWrapper(IAsyncResult result)
System.Net Verbose: 0 : [18252] Entering HttpWebRequest#37489757::EndGetRequestStream()
System.Net Error: 0 : [18252] Ausnahme in HttpWebRequest#37489757::EndGetRequestStream - Der Remoteserver hat einen Fehler zurückgegeben: (407) Proxyauthentifizierung erforderlich..
   bei System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
   bei System.Net.Connection.TunnelThroughProxyWrapper(IAsyncResult result)
System.Net.Http Error: 0 : [18252] Ausnahme in HttpClientHandler#55416443::SendAsync - Der Remoteserver hat einen Fehler zurückgegeben: (407) Proxyauthentifizierung erforderlich..
   bei System.Net.HttpWebRequest.EndGetRequestStream(IAsyncResult asyncResult, TransportContext& context)
   bei System.Net.Http.HttpClientHandler.GetRequestStreamCallback(IAsyncResult ar)
System.Net.Http Error: 0 : [18252] ConfigurableHttpClient#324444::SendAsync() - Fehler beim Senden von HttpRequestMessage#4032828. System.Net.Http.HttpRequestException: Fehler beim Senden der Anforderung. ---> System.Net.WebException: Der Remoteserver hat einen Fehler zurückgegeben: (407) Proxyauthentifizierung erforderlich.
   bei System.Net.HttpWebRequest.EndGetRequestStream(IAsyncResult asyncResult, TransportContext& context)
   bei System.Net.Http.HttpClientHandler.GetRequestStreamCallback(IAsyncResult ar)
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei Google.Apis.Http.ConfigurableMessageHandler.<SendAsync>d__69.MoveNext()
System.Net Information: 0 : [18252] ServicePoint#43245362::CloseConnectionGroupInternal(8198134)
System.Net Information: 0 : [18252] ServicePoint#43245362::CloseConnectionGroupHelper(connectionGroupName=8198134, closeInternal=True)
System.Net Information: 0 : [18252] ServicePoint#43245362::CloseConnectionGroupHelper, returning(False)
System.Net Information: 0 : [18252] ServicePoint#64828693::CloseConnectionGroupInternal(8198134)
System.Net Information: 0 : [18252] ServicePoint#64828693::CloseConnectionGroupHelper(connectionGroupName=8198134, closeInternal=True)
System.Net Information: 0 : [18252] ServicePoint#64828693::CloseConnectionGroupHelper, returning(False)
System.Net.Http Verbose: 0 : [16804] Entering HttpClientHandler#40799911::.ctor()
System.Net.Http Verbose: 0 : [16804] Exiting HttpClientHandler#40799911::.ctor() 
System.Net.Http Verbose: 0 : [16804] Entering StreamInterceptionHandler#31654880::.ctor()
System.Net.Http Verbose: 0 : [16804] Exiting StreamInterceptionHandler#31654880::.ctor() 
System.Net.Http Information: 0 : [16804] Associating StreamInterceptionHandler#31654880 with HttpClientHandler#40799911
System.Net.Http Verbose: 0 : [16804] Entering ConfigurableMessageHandler#16458469::.ctor()
System.Net.Http Verbose: 0 : [16804] Exiting ConfigurableMessageHandler#16458469::.ctor() 
System.Net.Http Information: 0 : [16804] Associating ConfigurableMessageHandler#16458469 with StreamInterceptionHandler#31654880
System.Net.Http Verbose: 0 : [16804] Entering ConfigurableHttpClient#13908493::.ctor(ConfigurableMessageHandler#16458469)
System.Net.Http Information: 0 : [16804] Associating ConfigurableHttpClient#13908493 with ConfigurableMessageHandler#16458469
System.Net.Http Verbose: 0 : [16804] Exiting ConfigurableHttpClient#13908493::.ctor() 
System.Net.Http Verbose: 0 : [16804] Entering ConfigurableHttpClient#13908493::.ctor(ConfigurableMessageHandler#16458469)
System.Net.Http Verbose: 0 : [16804] Exiting ConfigurableHttpClient#13908493::.ctor() 
System.Net.Http Verbose: 0 : [16804] Entering HttpClientHandler#58067579::.ctor()
System.Net.Http Verbose: 0 : [16804] Exiting HttpClientHandler#58067579::.ctor() 
System.Net.Http Verbose: 0 : [16804] Entering HttpClient#52846167::.ctor(HttpClientHandler#58067579)
System.Net.Http Information: 0 : [16804] Associating HttpClient#52846167 with HttpClientHandler#58067579
System.Net.Http Verbose: 0 : [16804] Exiting HttpClient#52846167::.ctor() 
System.Net.Http Verbose: 0 : [16804] Entering HttpClient#52846167::.ctor(HttpClientHandler#58067579)
System.Net.Http Verbose: 0 : [16804] Exiting HttpClient#52846167::.ctor() 
System.Net.Http Verbose: 0 : [16804] Entering HttpRequestMessage#5853458::.ctor(Method: GET, Uri: 'https://openidconnect.googleapis.com/v1/userinfo')
System.Net.Http Verbose: 0 : [16804] Exiting HttpRequestMessage#5853458::.ctor() 
System.Net.Http Verbose: 0 : [48416] Entering HttpRequestMessage#50445525::.ctor(Method: POST, Uri: 'https://oauth2.googleapis.com/token')
System.Net.Http Verbose: 0 : [48416] Exiting HttpRequestMessage#50445525::.ctor() 
System.Net.Http Verbose: 0 : [48416] Entering FormUrlEncodedContent#54089615::.ctor()
System.Net.Http Verbose: 0 : [48416] Exiting FormUrlEncodedContent#54089615::.ctor() 
System.Net.Http Information: 0 : [48416] Associating HttpRequestMessage#50445525 with FormUrlEncodedContent#54089615
System.Net.Http Verbose: 0 : [48416] Entering ConfigurableHttpClient#13908493::SendAsync(HttpRequestMessage#50445525: Method: POST, RequestUri: 'https://oauth2.googleapis.com/token', Version: 1.1, Content: System.Net.Http.FormUrlEncodedContent, Headers:
{
  Content-Type: application/x-www-form-urlencoded
})
System.Net.Http Verbose: 0 : [48416] Entering HttpClientHandler#40799911::SendAsync(HttpRequestMessage#50445525)
System.Net Verbose: 0 : [48416] Entering HttpWebRequest#40126858::HttpWebRequest(https://oauth2.googleapis.com/token#-1749135709)
System.Net Verbose: 0 : [48416] Exiting HttpWebRequest#40126858::HttpWebRequest() 
System.Net Verbose: 0 : [48416] Entering HttpWebRequest#40126858::HttpWebRequest(uri: 'https://oauth2.googleapis.com/token', connectionGroupName: '40799911')
System.Net Verbose: 0 : [48416] Exiting HttpWebRequest#40126858::HttpWebRequest() 
System.Net.Http Information: 0 : [48416] Associating HttpRequestMessage#50445525 with HttpWebRequest#40126858
System.Net Information: 0 : [48416] Associating HttpWebRequest#40126858 with ServicePoint#64828693
System.Net.Http Verbose: 0 : [48416] Exiting HttpClientHandler#40799911::SendAsync() 	-> Task`1#49117909
System.Net.Http Verbose: 0 : [48416] Exiting ConfigurableHttpClient#13908493::SendAsync() 	-> Task`1#10278272
System.Net Verbose: 0 : [66412] Entering HttpWebRequest#40126858::BeginGetRequestStream()
System.Net Information: 0 : [66412] Associating Connection#12114023 with HttpWebRequest#40126858
System.Net Verbose: 0 : [66412] Entering HttpWebRequest#4611808::HttpWebRequest(http://pxvip02.a.domain.:8080/#1479616988)
System.Net Verbose: 0 : [66412] Exiting HttpWebRequest#4611808::HttpWebRequest() 
System.Net Information: 0 : [66412] Associating HttpWebRequest#4611808 with ServicePoint#43245362
System.Net Verbose: 0 : [66412] Entering HttpWebRequest#4611808::BeginGetResponse()
System.Net Information: 0 : [66412] Associating Connection#11291881 with HttpWebRequest#4611808
System.Net Verbose: 0 : [66412] Exiting HttpWebRequest#4611808::BeginGetResponse() 	-> ContextAwareResult#57744263
System.Net Verbose: 0 : [66412] Exiting HttpWebRequest#40126858::BeginGetRequestStream() 	-> ContextAwareResult#42128381
System.Net Information: 0 : [18252] Connection#11291881 - Die Verbindung von 10.80.6.44:59099 mit 10.10.20.9:8080 wurde hergestellt.
System.Net Information: 0 : [18252] Associating HttpWebRequest#4611808 with ConnectStream#61198641
System.Net Information: 0 : [18252] HttpWebRequest#4611808 - Request: CONNECT oauth2.googleapis.com:443 HTTP/1.1

System.Net Information: 0 : [18252] ConnectStream#61198641 - Header
{
User-Agent: google-api-dotnet-client/1.57.0.0 (gzip)
Host: oauth2.googleapis.com
} werden gesendet.
System.Net Information: 0 : [66180] Connection#11291881 - Empfangene Statusleiste: Version = 1.1, StatusCode = 407, StatusDescription = Proxy Authentication Required.
System.Net Information: 0 : [66180] Connection#11291881 - Header
{
Proxy-Authenticate: NEGOTIATE,NTLM,BASIC realm="myrealm"
Cache-Control: no-cache
Pragma: no-cache
Content-Type: text/html; charset=utf-8
Proxy-Connection: close
Connection: close
Content-Length: 797
} wurden empfangen.
System.Net Information: 0 : [66180] ConnectStream#62130720::ConnectStream(Es wurden 797 Bytes gepuffert.)
System.Net Information: 0 : [66180] Associating HttpWebRequest#4611808 with ConnectStream#62130720
System.Net Information: 0 : [66180] Associating HttpWebRequest#4611808 with HttpWebResponse#4593845
System.Net Information: 0 : [66180] AcquireCredentialsHandle(package  = Negotiate, intent   = Outbound, authdata = System.Net.SafeSspiAuthDataHandle)
System.Net Information: 0 : [66180] InitializeSecurityContext(credential = System.Net.SafeFreeCredential_SECURITY, context = (null), targetName = HTTP/pxvip02.a.domain, inFlags = Delegate, MutualAuth, Connection)
System.Net Information: 0 : [66180] InitializeSecurityContext(Anzahl von In-Buffers = 0, Länge von Out-Buffer = 40, zurückgegebener Code = ContinueNeeded).
System.Net Warning: 0 : [66180] HttpWebRequest#4611808::() - Anforderung wird erneut übertragen.
System.Net Information: 0 : [66180] Associating HttpWebRequest#4611808 with ServicePoint#43245362
System.Net Information: 0 : [66180] Associating Connection#42147222 with HttpWebRequest#4611808
System.Net Information: 0 : [66180] Connection#42147222 - Die Verbindung von 10.80.6.44:59100 mit 10.10.20.9:8080 wurde hergestellt.
System.Net Information: 0 : [66180] Associating HttpWebRequest#4611808 with ConnectStream#19210254
System.Net Information: 0 : [66180] HttpWebRequest#4611808 - Request: CONNECT oauth2.googleapis.com:443 HTTP/1.1

System.Net Information: 0 : [66180] ConnectStream#19210254 - Header
{
User-Agent: google-api-dotnet-client/1.57.0.0 (gzip)
Proxy-Authorization: Negotiate TlRMTVNTUAABAAAAl4II4gAAAAAAAAAAAAAAAAAAAAAKAGFKAAAADw==
Host: oauth2.googleapis.com
} werden gesendet.
System.Net Information: 0 : [18252] Connection#42147222 - Empfangene Statusleiste: Version = 1.1, StatusCode = 407, StatusDescription = Proxy Authentication Required.
System.Net Information: 0 : [18252] Connection#42147222 - Header
{
Pragma: no-cache
Proxy-Connection: Keep-Alive
Connection: Keep-Alive
Content-Length: 814
Cache-Control: no-cache
Content-Type: text/html; charset=utf-8
Proxy-Authenticate: NEGOTIATE TlRMTVNTUAACAAAABgAGADgAAAAVAong3lmTaA35kycAAAAAAAAAAJQAlAA+AAAABQCTCAAAAA9aAFQAQgACAAYAWgBUAEIAAQAOAFAAWABGAEYATQAyADQABAAuAFoAVABCAC4ASQBDAEIALgBDAE8ATQBNAEUAUgBaAEIAQQBOAEsALgBDAE8ATQADAD4AcAB4AGYAZgBtADIANAAuAHoAdABiAC4AaQBjAGIALgBjAG8AbQBtAGUAcgB6AGIAYQBuAGsALgBjAG8AbQAAAAAA
} wurden empfangen.
System.Net Information: 0 : [18252] ConnectStream#45943393::ConnectStream(Es wurden 814 Bytes gepuffert.)
System.Net Information: 0 : [18252] Associating HttpWebRequest#4611808 with ConnectStream#45943393
System.Net Information: 0 : [18252] Associating HttpWebRequest#4611808 with HttpWebResponse#36633745
System.Net Information: 0 : [18252] InitializeSecurityContext(credential = System.Net.SafeFreeCredential_SECURITY, context = a995f48:a968f98, targetName = HTTP/pxvip02.a.domain, inFlags = Delegate, MutualAuth, Connection)
System.Net Information: 0 : [18252] InitializeSecurityContext(Anzahl von In-Buffers = 1, Länge von Out-Buffer = 504, zurückgegebener Code = OK).
System.Net Warning: 0 : [18252] HttpWebRequest#4611808::() - Anforderung wird erneut übertragen.
System.Net Information: 0 : [18252] Associating HttpWebRequest#4611808 with ServicePoint#43245362
System.Net Information: 0 : [18252] Associating Connection#42147222 with HttpWebRequest#4611808
System.Net Information: 0 : [18252] Associating HttpWebRequest#4611808 with ConnectStream#50223079
System.Net Information: 0 : [18252] HttpWebRequest#4611808 - Request: CONNECT oauth2.googleapis.com:443 HTTP/1.1

System.Net Information: 0 : [18252] ConnectStream#50223079 - Header
{
User-Agent: google-api-dotnet-client/1.57.0.0 (gzip)
Proxy-Authorization: Negotiate TlRMTVNTUAADAAAAGAAYAHYAAABaAVoBjgAAAAAAAABYAAAADgAOAFgAAAAQABAAZgAAABAAEADoAQAAFQKI4goAYUoAAAAPay0F30ofDTQbrxj4uSt4dWMAYgAyAGwAbwBiAHIAQQBQADkAMgA3ADkANQAyAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADhNEwol71x0liyEsaGbzRIBAQAAAAAAANCVtmCbE9kBw84YXhLDbEZLDp03AgAGAFoAVABCAAEADgBQAFgARgBGAE0AMgA0AAQALgBaAFQAQgAuAEkAQwBCAC4AQwBPAE0ATQBFAFIAWgBCAEEATgBLAC4AQwBPAE0AAwA+AHAAeABmAGYAbQAyADQALgB6AHQAYgAuAGkAYwBiAC4AYwBvAG0AbQBlAHIAegBiAGEAbgBrAC4AYwBvAG0ACAAwADAAAAAAAAAAAAAAAAAgAABADAG42Xfm+XxzC2p2aMxtS+0HUfhw2iIHgfXTWXPnqwoAEAAAAAAAAAAAAAAAAAAAAAAACQBKAEgAVABUAFAALwBwAHgAdgBpAHAAMAAyAC4AaQBuAHQAcgBhAG4AZQB0AC4AYwBvAG0AbQBlAHIAegBiAGEAbgBrAC4AYwBvAG0AAAAAANggk9mDzIsTA6uI2Xzt5LVqLebi
Host: oauth2.googleapis.com
} werden gesendet.
System.Net Information: 0 : [66180] Connection#42147222 - Empfangene Statusleiste: Version = 1.1, StatusCode = 407, StatusDescription = Proxy Authentication Required.
System.Net Information: 0 : [66180] Connection#42147222 - Header
{
Proxy-Authenticate: NEGOTIATE,NTLM
Cache-Control: no-cache
Pragma: no-cache
Content-Type: text/html; charset=utf-8
Proxy-Connection: close
Connection: close
Content-Length: 847
} wurden empfangen.
System.Net Information: 0 : [66180] ConnectStream#11510502::ConnectStream(Es wurden 847 Bytes gepuffert.)
System.Net Information: 0 : [66180] Associating HttpWebRequest#4611808 with ConnectStream#11510502
System.Net Information: 0 : [66180] Associating HttpWebRequest#4611808 with HttpWebResponse#6076874
System.Net Information: 0 : [66180] ServicePoint#43245362::ReleaseConnectionGroup(/.NET/NetClasses/HttpWebRequest/CONNECT__Group$$/(2)S>I>System.Net.NegotiateClient/3)
System.Net Information: 0 : [66180] ServicePoint#43245362::ReleaseConnectionGroup, returning(true)
System.Net Verbose: 0 : [66180] Entering ConnectStream#11510502::Read()
System.Net Verbose: 0 : [66180] Data from ConnectStream#11510502::Read
System.Net Verbose: 0 : [66180] 00000000 : 3C 48 54 4D 4C 3E 3C 48-45 41 44 3E 0D 0A 3C 54 : <HTML><HEAD>..<T
System.Net Verbose: 0 : [66180] 00000010 : 49 54 4C 45 3E 41 63 63-65 73 73 20 44 65 6E 69 : ITLE>Access Deni
System.Net Verbose: 0 : [66180] 00000020 : 65 64 3C 2F 54 49 54 4C-45 3E 0D 0A 3C 2F 48 45 : ed</TITLE>..</HE
System.Net Verbose: 0 : [66180] 00000030 : 41 44 3E 0D 0A 3C 42 4F-44 59 3E 0D 0A 3C 62 6F : AD>..<BODY>..<bo
System.Net Verbose: 0 : [66180] 00000040 : 64 79 20 62 67 63 6F 6C-6F 72 3D 22 23 46 46 46 : dy bgcolor="#FFF
System.Net Verbose: 0 : [66180] 00000050 : 46 45 41 22 3E 0D 0A 3C-62 6C 6F 63 6B 71 75 6F : FEA">..<blockquo
System.Net Verbose: 0 : [66180] 00000060 : 74 65 3E 0D 0A 3C 54 41-42 4C 45 20 62 6F 72 64 : te>..<TABLE bord
System.Net Verbose: 0 : [66180] 00000070 : 65 72 3D 30 20 63 65 6C-6C 50 61 64 64 69 6E 67 : er=0 cellPadding
System.Net Verbose: 0 : [66180] 00000080 : 3D 31 20 77 69 64 74 68-3D 22 38 30 25 22 3E 0D : =1 width="80%">.
System.Net Verbose: 0 : [66180] 00000090 : 0A 3C 54 52 3E 3C 54 44-3E 0D 0A 3C 46 4F 4E 54 : .<TR><TD>..<FONT
System.Net Verbose: 0 : [66180] 000000A0 : 20 66 61 63 65 3D 22 56-65 72 64 61 6E 61 22 3E :  face="Verdana">
System.Net Verbose: 0 : [66180] 000000B0 : 0D 0A 3C 62 69 67 3E 41-63 63 65 73 73 20 44 65 : ..<big>Access De
System.Net Verbose: 0 : [66180] 000000C0 : 6E 69 65 64 20 28 61 75-74 68 65 6E 74 69 63 61 : nied (authentica
System.Net Verbose: 0 : [66180] 000000D0 : 74 69 6F 6E 5F 66 61 69-6C 65 64 29 3C 2F 62 69 : tion_failed)</bi
System.Net Verbose: 0 : [66180] 000000E0 : 67 3E 0D 0A 3C 42 52 3E-0D 0A 3C 42 52 3E 0D 0A : g>..<BR>..<BR>..
System.Net Verbose: 0 : [66180] 000000F0 : 3C 2F 46 4F 4E 54 3E 0D-0A 3C 2F 54 44 3E 3C 2F : </FONT>..</TD></
System.Net Verbose: 0 : [66180] 00000100 : 54 52 3E 0D 0A 3C 54 52-3E 3C 54 44 3E 0D 0A 3C : TR>..<TR><TD>..<
System.Net Verbose: 0 : [66180] 00000110 : 46 4F 4E 54 20 66 61 63-65 3D 22 56 65 72 64 61 : FONT face="Verda
System.Net Verbose: 0 : [66180] 00000120 : 6E 61 22 3E 0D 0A 59 6F-75 72 20 63 72 65 64 65 : na">..Your crede
System.Net Verbose: 0 : [66180] 00000130 : 6E 74 69 61 6C 73 20 63-6F 75 6C 64 20 6E 6F 74 : ntials could not
System.Net Verbose: 0 : [66180] 00000140 : 20 62 65 20 61 75 74 68-65 6E 74 69 63 61 74 65 :  be authenticate
System.Net Verbose: 0 : [66180] 00000150 : 64 3A 20 22 47 65 6E 65-72 61 6C 20 61 75 74 68 : d: "General auth
System.Net Verbose: 0 : [66180] 00000160 : 65 6E 74 69 63 61 74 69-6F 6E 20 66 61 69 6C 75 : entication failu
System.Net Verbose: 0 : [66180] 00000170 : 72 65 20 64 75 65 20 74-6F 20 62 61 64 20 75 73 : re due to bad us
System.Net Verbose: 0 : [66180] 00000180 : 65 72 20 49 44 20 6F 72-20 61 75 74 68 65 6E 74 : er ID or authent
System.Net Verbose: 0 : [66180] 00000190 : 69 63 61 74 69 6F 6E 20-74 6F 6B 65 6E 2E 22 2E : ication token.".
System.Net Verbose: 0 : [66180] 000001A0 : 20 59 6F 75 20 77 69 6C-6C 20 6E 6F 74 20 62 65 :  You will not be
System.Net Verbose: 0 : [66180] 000001B0 : 20 70 65 72 6D 69 74 74-65 64 20 61 63 63 65 73 :  permitted acces
System.Net Verbose: 0 : [66180] 000001C0 : 73 20 75 6E 74 69 6C 20-79 6F 75 72 20 63 72 65 : s until your cre
System.Net Verbose: 0 : [66180] 000001D0 : 64 65 6E 74 69 61 6C 73-20 63 61 6E 20 62 65 20 : dentials can be 
System.Net Verbose: 0 : [66180] 000001E0 : 76 65 72 69 66 69 65 64-2E 0D 0A 3C 2F 46 4F 4E : verified...</FON
System.Net Verbose: 0 : [66180] 000001F0 : 54 3E 0D 0A 3C 2F 54 44-3E 3C 2F 54 52 3E 0D 0A : T>..</TD></TR>..
System.Net Verbose: 0 : [66180] 00000200 : 3C 54 52 3E 3C 54 44 3E-0D 0A 3C 46 4F 4E 54 20 : <TR><TD>..<FONT 
System.Net Verbose: 0 : [66180] 00000210 : 66 61 63 65 3D 22 56 65-72 64 61 6E 61 22 3E 0D : face="Verdana">.
System.Net Verbose: 0 : [66180] 00000220 : 0A 54 68 69 73 20 69 73-20 74 79 70 69 63 61 6C : .This is typical
System.Net Verbose: 0 : [66180] 00000230 : 6C 79 20 63 61 75 73 65-64 20 62 79 20 61 6E 20 : ly caused by an 
System.Net Verbose: 0 : [66180] 00000240 : 69 6E 63 6F 72 72 65 63-74 20 75 73 65 72 6E 61 : incorrect userna
System.Net Verbose: 0 : [66180] 00000250 : 6D 65 20 61 6E 64 2F 6F-72 20 70 61 73 73 77 6F : me and/or passwo
System.Net Verbose: 0 : [66180] 00000260 : 72 64 2C 20 62 75 74 20-63 6F 75 6C 64 20 61 6C : rd, but could al
System.Net Verbose: 0 : [66180] 00000270 : 73 6F 20 62 65 20 63 61-75 73 65 64 20 62 79 20 : so be caused by 
System.Net Verbose: 0 : [66180] 00000280 : 6E 65 74 77 6F 72 6B 20-70 72 6F 62 6C 65 6D 73 : network problems
System.Net Verbose: 0 : [66180] 00000290 : 2E 0D 0A 3C 2F 46 4F 4E-54 3E 0D 0A 3C 2F 54 44 : ...</FONT>..</TD
System.Net Verbose: 0 : [66180] 000002A0 : 3E 3C 2F 54 52 3E 0D 0A-3C 54 52 3E 3C 54 44 3E : ></TR>..<TR><TD>
System.Net Verbose: 0 : [66180] 000002B0 : 0D 0A 3C 46 4F 4E 54 20-66 61 63 65 3D 22 56 65 : ..<FONT face="Ve
System.Net Verbose: 0 : [66180] 000002C0 : 72 64 61 6E 61 22 20 53-49 5A 45 3D 32 3E 0D 0A : rdana" SIZE=2>..
System.Net Verbose: 0 : [66180] 000002D0 : 3C 42 52 3E 0D 0A 46 6F-72 20 61 73 73 69 73 74 : <BR>..For assist
System.Net Verbose: 0 : [66180] 000002E0 : 61 6E 63 65 2C 20 63 6F-6E 74 61 63 74 20 79 6F : ance, contact yo
System.Net Verbose: 0 : [66180] 000002F0 : 75 72 20 54 65 63 68 20-44 65 73 6B 20 2E 20 50 : ur Tech Desk . P
System.Net Verbose: 0 : [66180] 00000300 : 58 46 46 4D 32 34 0D 0A-3C 2F 46 4F 4E 54 3E 0D : XFFM24..</FONT>.
System.Net Verbose: 0 : [66180] 00000310 : 0A 3C 2F 54 44 3E 3C 2F-54 52 3E 0D 0A 3C 2F 54 : .</TD></TR>..</T
System.Net Verbose: 0 : [66180] 00000320 : 41 42 4C 45 3E 0D 0A 3C-2F 62 6C 6F 63 6B 71 75 : ABLE>..</blockqu
System.Net Verbose: 0 : [66180] 00000330 : 6F 74 65 3E 0D 0A 3C 2F-46 4F 4E 54 3E 0D 0A 3C : ote>..</FONT>..<
System.Net Verbose: 0 : [66180] 00000340 : 2F 42 4F 44 59 3E 3C 2F-48 54 4D 4C 3E 0D 0A    : /BODY></HTML>..
System.Net Verbose: 0 : [66180] Exiting ConnectStream#11510502::Read() 	-> Int32#847
System.Net Verbose: 0 : [66180] Entering ConnectStream#11510502::Read()
System.Net Verbose: 0 : [66180] Exiting ConnectStream#11510502::Read() 	-> Int32#0
System.Net Verbose: 0 : [66180] Entering HttpWebRequest#4611808::EndGetResponse()
System.Net Error: 0 : [66180] Ausnahme in HttpWebRequest#4611808::EndGetResponse - Der Remoteserver hat einen Fehler zurückgegeben: (407) Proxyauthentifizierung erforderlich..
System.Net Error: 0 : [66180] Ausnahme in HttpWebRequest#40126858:: - Der Remoteserver hat einen Fehler zurückgegeben: (407) Proxyauthentifizierung erforderlich..
   bei System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
   bei System.Net.Connection.TunnelThroughProxyWrapper(IAsyncResult result)
System.Net Verbose: 0 : [66180] Entering HttpWebRequest#40126858::EndGetRequestStream()
System.Net Error: 0 : [66180] Ausnahme in HttpWebRequest#40126858::EndGetRequestStream - Der Remoteserver hat einen Fehler zurückgegeben: (407) Proxyauthentifizierung erforderlich..
   bei System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult)
   bei System.Net.Connection.TunnelThroughProxyWrapper(IAsyncResult result)
System.Net.Http Error: 0 : [66180] Ausnahme in HttpClientHandler#40799911::SendAsync - Der Remoteserver hat einen Fehler zurückgegeben: (407) Proxyauthentifizierung erforderlich..
   bei System.Net.HttpWebRequest.EndGetRequestStream(IAsyncResult asyncResult, TransportContext& context)
   bei System.Net.Http.HttpClientHandler.GetRequestStreamCallback(IAsyncResult ar)
System.Net.Http Error: 0 : [66180] ConfigurableHttpClient#13908493::SendAsync() - Fehler beim Senden von HttpRequestMessage#50445525. System.Net.Http.HttpRequestException: Fehler beim Senden der Anforderung. ---> System.Net.WebException: Der Remoteserver hat einen Fehler zurückgegeben: (407) Proxyauthentifizierung erforderlich.
   bei System.Net.HttpWebRequest.EndGetRequestStream(IAsyncResult asyncResult, TransportContext& context)
   bei System.Net.Http.HttpClientHandler.GetRequestStreamCallback(IAsyncResult ar)
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei Google.Apis.Http.ConfigurableMessageHandler.<SendAsync>d__69.MoveNext()
System.Net Information: 0 : [66180] ServicePoint#43245362::CloseConnectionGroupInternal(58067579)
System.Net Information: 0 : [66180] ServicePoint#43245362::CloseConnectionGroupHelper(connectionGroupName=58067579, closeInternal=True)
System.Net Information: 0 : [66180] ServicePoint#43245362::CloseConnectionGroupHelper, returning(False)
System.Net Information: 0 : [66180] ServicePoint#64828693::CloseConnectionGroupInternal(58067579)
System.Net Information: 0 : [66180] ServicePoint#64828693::CloseConnectionGroupHelper(connectionGroupName=58067579, closeInternal=True)
System.Net Information: 0 : [66180] ServicePoint#64828693::CloseConnectionGroupHelper, returning(False)
System.Net Information: 0 : [63704] ServicePoint#43245362 - ConnectionGroup#37121646 - Geschlossen, da im Leerlauf.
System.Net Information: 0 : [63704] ServicePoint#43245362::ReleaseConnectionGroup(/.NET/NetClasses/HttpWebRequest/CONNECT__Group$$/(1)S>I>)
System.Net Information: 0 : [63704] ServicePoint#43245362::ReleaseConnectionGroup, returning(true)
System.Net Information: 0 : [63704] ServicePoint#43245362 - ConnectionGroup#45592480 - Geschlossen, da im Leerlauf.
System.Net Information: 0 : [63704] ServicePoint#43245362::ReleaseConnectionGroup(/.NET/NetClasses/HttpWebRequest/CONNECT__Group$$/(1)S>I>System.Net.NegotiateClient/2)
System.Net Information: 0 : [63704] ServicePoint#43245362::ReleaseConnectionGroup, returning(false)
System.Net Information: 0 : [63704] ServicePoint#43245362 - ConnectionGroup#57352375 - Geschlossen, da im Leerlauf.
System.Net Information: 0 : [63704] ServicePoint#43245362::ReleaseConnectionGroup(/.NET/NetClasses/HttpWebRequest/CONNECT__Group$$/(2)S>I>)
System.Net Information: 0 : [63704] ServicePoint#43245362::ReleaseConnectionGroup, returning(true)
System.Net Information: 0 : [63704] ServicePoint#43245362 - Geschlossen, da im Leerlauf.
System.Net Information: 0 : [63704] ServicePoint#43245362 - ConnectionGroup#2637164 - Geschlossen, da im Leerlauf.
System.Net Information: 0 : [63704] ServicePoint#43245362::ReleaseConnectionGroup(/.NET/NetClasses/HttpWebRequest/CONNECT__Group$$/(2)S>I>System.Net.NegotiateClient/3)
System.Net Information: 0 : [63704] ServicePoint#43245362::ReleaseConnectionGroup, returning(false)

**Comment by chipmunk2000** [comment]
When starting it tells me loading projects failed n ... again a 407 error

contents of systen.net.log ==> I had to change the hostnames and IP'S and realm for obvious reasons :-)
[mysystem.net.log](https://github.com/GoogleCloudPlatform/iap-desktop/files/10259032/mysystem.net.log)

**Comment by jpassing** [comment]
The log file that you attached looks ok: It shows several successful proxy authentications (using NTLM) and IAP Desktop successfully completed more than a dozen API requests. 

But your original comment included a different log, and that one showed several NTLM proxy authentication errors. How do those two logs relate?

**Comment by chipmunk2000** [comment]
The first log was from the normal logging, the new log is from the trial with the new config file.

it tries to lookup the existing projects, that I had configured previously, some of them show up as inaccessible.
I havent tried in this attempt to add a new project, I have just started it up and had already all those 407 Proxy errors.



If you want me tor try it with adding projects, i can repeat, I thought it was sufficient already as the 407 errors were present


Cheers Olaf

**Comment by jpassing** [comment]
If I understand correctly, then the 407 errors are essentially occurring intermittently -- sometimes the app starts properly, but fails to load some projects, sometimes it fails to load other projects, and sometimes it even fails during startup. Is that accurate?

I'm having some trouble analyzing the logs because they're pretty heavily redacted. One thing that's visible though is that there are frequent NTLM handshakes taking place. That's no surprise given that by default, each new proxy connection requires a new NTLM handshake -- but it's possible that the frequency contributes to (or even causes) the issue. It's also possible that the proxy server fails some of the requests because of it.

Since [version 2.30](https://github.com/GoogleCloudPlatform/iap-desktop/releases/tag/2.30.878), IAP Desktop loads projects in parallel (to improve performance) while previously, it loaded them sequentially. If NTLM proxy authentication is failing intermittently, then it would make sense that the errors are much more frequent now than they were with previous versions.

**Comment by chipmunk2000** [comment]
They are not that heavily redacted, just the company replaced by soething else, realm replaced with something else username replaced by x‘s and the ip address of proxy changed and name of issuer in certificate changed If i can make some other tests let me knowBut some projects that were included do work , while others dont and adding any simply does not work so that points in your direction of parallelismCheers OlafAm 20.12.2022 um 13:57 schrieb Johannes Passing ***@***.***>:﻿
If I understand correctly, then the 407 errors are essentially occurring intermittently -- sometimes the app starts properly, but fails to load some projects, sometimes it fails to load other projects, and sometimes it even fails during startup. Is that accurate?
I'm having some trouble analyzing the logs because they're pretty heavily redacted. One thing that's visible though is that there are frequent NTLM handshakes taking place. That's no surprise given that by default, each new proxy connection requires a new NTLM handshake -- but it's possible that the frequency contributes to (or even causes) the issue. It's also possible that the proxy server fails some of the requests because of it.
Since version 2.30, IAP Desktop loads projects in parallel (to improve performance) while previously, it loaded them sequentially. If NTLM proxy authentication is failing intermittently, then it would make sense that the errors are much more frequent now than they were with previous versions.

**Comment by jpassing** [comment]
It would be great if you could create and share one more trace file:

1. Delete/Rename the existing `systen.net.log`
2. Start IAP Desktop and use it until the 407 error occurs
3. Send the log to iap-desktop@google.com (to keep it private)

As indicated in my previous comment, it seems to me that IAP Desktop isn't really doing anything wrong -- but the way it manages connections (opening many connections in parallel and closing them as soon as they're not needed anymore) might be incompatible with the proxy server's quotas/limits/policies. If that assessment is right, then the fix would be to change IAP Desktop to use fewer connections, and to pool them for longer. That's not a trivial change though.

**Comment by jpassing** [comment]
I implemented two changes that I hope will help:

1. The app now pools and reuses TCP connections more aggressively. That should help bring down the overall number of connections used, and reduce proxy load.
2. You can now set a limit on how many connections the app will open per endpoint (the default is 16). That lets you further reduce proxy load.

If my previous assessment is right, then limiting the number of connections should fix the issue, or at least make those 407 errors much less frequent.

It would be great if you could give the following a try:

1. Download and install the attached build (it's a signed build of the latest `master` branch):
[IapDesktop-2.33.976-x86-Release.msi.zip](https://github.com/GoogleCloudPlatform/iap-desktop/files/10287261/IapDesktop-2.33.976-x86-Release.msi.zip)
2. Run the following command to set a limit on 2 connections per endpoint:
   ```
   reg add "HKCU\SOFTWARE\Google\IapDesktop\1.0\Application" /v ConnectionLimit /t REG_DWORD /d 2
   ```
3. (Re)start the app and see if it makes a difference.

You can keep using this build, or downgrade to a previous version. Previous versions will simply ignore this new registry value.

Thanks for your cooperation!

**Comment by chipmunk2000** [comment]
Hi Johannes, it is working again, a bit slow now when selecting projects, but working again


tried it as well with increased  Connectionlimit, but it fails immediatly when Raising the connectionlimit to eg 4


with two it is working fine

Thanks Olaf

**Comment by jpassing** [comment]
Great, glad to hear that it works now. The performance degradation is expected, but should only affect the Project Explorer. In particular, RDP and SSH performance should be unaffected.

For the next release, I'm going to expose the connection limit in the Options dialog. I'll keep the issue open until then.

**Comment by jootuom** [comment]
I ran into something similar or it's another issue that warrants its own \#, but I'll post here.

We also have an (Kerberos) authenticating proxy (Squid) for auditability purposes.

iap-desktop is unable to connect to VM instances and says "Connecting to VM instance failed", Details -> "WebException: The remote server returned an error: (407) Proxy Authentication Required."

We do allowlisting on the proxy (with and without authentication depending on case) so I permitted "tunnel.cloudproxy.app" (according to docs) after authentication and it's still not working. It looks to me like iap-desktop is not using credentials properly. Proxy logs only show the initial 407 pre-auth.

Since iap-desktop appears to be a .NET app, as a sidenote, according to our experience, at least PowerShell requires some manual massaging so that Kerberos authentication works for proxies, this issue might be similar...

This is what we have to do in PowerShell sessions:
`(New-Object -TypeName System.Net.WebClient).Proxy.Credentials = [System.Net.CredentialCache]::DefaultNetworkCredentials`

**Comment by jpassing** [comment]
If I understand correctly, then launching IAP Desktop, signing in, and listing VMs in Project Explorer all works fine -- but once you connect to a VM, you get a 407 error. Is that correct? If so, then I don't think proxy authentication is failing in general -- instead, it seems that it works fine for most endpoints, but not for `tunnel.cloudproxy.app`.

Connecting to a VM (or rather, establishing an IAP-TCP tunnel) uses the `tunnel.cloudproxy.app` endpoint. This endpoint differs from other API endpoint in (a) its domain (`.cloudproxy.app` vs `.googleapis.com`) and in that it uses WebSockets (instead of plain REST). Some proxies disallow WebSockets, or apply some special policies to WebSockets connections -- could that be the case here?

It might be worth checking if opening Cloud Shell from within the Cloud Console works. Cloud Shell also uses WebSockets, and if WebSockets are disallowed, opening Cloud Shell should fail too.

> This is what we have to do in PowerShell sessions:
> `(New-Object -TypeName System.Net.WebClient).Proxy.Credentials = [System.Net.CredentialCache]::DefaultNetworkCredentials`

Yes, this is more or less what IAP Desktop uses internally as well. Note that you have to specify your credentials under **Tools > Options > Network**, the app currently doesn't pick up your "default" proxy credentials.

To further diagnose the issue, it would be great if you could enable and share diagnostics logs as [described in a previous comment](#issuecomment-1354954704). These logs might contain more cues about why proxy authentication is failing. 

If possible, please send the log to iap-desktop@google.com instead of posting it here.

(FWIW, I don't think this issue is related to the issue chipmunk2000 encountered as that was specifically about NTLM proxy authentication)

**Comment by jpassing** [comment]
The connection pool-related changes a now available in [release 2.34.988](https://github.com/GoogleCloudPlatform/iap-desktop/releases/tag/2.34.988). I also extended the Options dialog so that you can change the limit in the UI instead of having to do it in the registry:

![image](https://user-images.githubusercontent.com/5724941/212201980-19be4f0d-7c2c-4560-8612-063863da49f2.png)

I'll close this issue as I think the NTLM-related problem is now resolved. 

@jootuom, I haven't heard back from you so I'm not sure what the status of the Kerberos-related problem is. If it persists, it would be great if you could open a separate issue. Thanks.
